Jenkinsfile

```
@@ -0,0 +1,113 @@
+                    echo env.TAG_NAME
+
+                    if (!(VERSION ==~ RELEASE_REGEX || VERSION ==~ /.*-SNAPSHOT$/)) {
+                        echo 'Version:'
+                        echo VERSION
+                        error 'The version declaration is invalid. It is neither a release nor a snapshot.'
+                    }
+                }
+                script {
+                    sh 'uv sync --locked --only-group gdal-build-dependencies'
+                    sh 'uv sync --locked'
+                    sh 'uv run pybabel compile -d sketch_map_tool/translations'
+                    sh 'wget --quiet -P weights https://sketch-map-tool.heigit.org/weights/SMT-OSM.pt'
+                    sh 'wget --quiet -P weights https://sketch-map-tool.heigit.org/weights/SMT-ESRI.pt'
+                    sh 'wget --quiet -P weights https://sketch-map-tool.heigit.org/weights/SMT-CLS.pt'
+                }
+            }
+            post {
+                failure {
+                  rocket_buildfail()
+                }
+            }
+        }
+
+        stage('Build Frontend') {
+            steps {
+                nodejs(nodeJSInstallationName: 'NodeJS 18') {
+                    sh 'npm install'
+                    sh 'npm run build'
+                }
+            }
+            post {
+                failure {
+                  rocket_buildfail()
+                }
+            }
+        }
+
+        stage('Test') {
+            environment {
+                VIRTUAL_ENV="${WORKSPACE}/.venv"
+                PATH="${VIRTUAL_ENV}/bin:${PATH}"
+            }
+            steps {
+                script {
+                    // run pytest
+                    sh 'VCR_RECORD_MODE=none pytest --cov=sketch_map_tool --cov-report=xml tests'
+                    // run static analysis with sonar-scanner
+                    def scannerHome = tool 'SonarScanner 4'
+                    withSonarQubeEnv('sonarcloud GIScience/ohsome') {
+                        SONAR_CLI_PARAMETER =
+                            "-Dsonar.python.coverage.reportPaths=${WORKSPACE}/coverage.xml " +
+                            "-Dsonar.projectVersion=${VERSION}"
+                        if (env.CHANGE_ID) {
+                            SONAR_CLI_PARAMETER += ' ' +
+                                "-Dsonar.pullrequest.key=${env.CHANGE_ID} " +
+                                "-Dsonar.pullrequest.branch=${env.CHANGE_BRANCH} " +
+                                "-Dsonar.pullrequest.base=${env.CHANGE_TARGET}"
+                        } else {
+                            SONAR_CLI_PARAMETER += ' ' +
+                                "-Dsonar.branch.name=${env.BRANCH_NAME}"
+                        }
+                        sh "${scannerHome}/bin/sonar-scanner " + SONAR_CLI_PARAMETER
+                    }
+                    // run other static code analysis
+                    sh 'ruff format --check --diff .'
+                    sh 'ruff check .'
```

We could also run all pre-commit hooks instead. This also includes linters for front-end.
```suggestion
                    sh 'pre-commit install && pre-commit run --all'
```